web/src/components/storage/DASDFormatProgress.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React from "react";
+import { Progress, Stack } from "@patternfly/react-core";
+import { Popup } from "~/components/core";
+import { _ } from "~/i18n";
+import { useDASDDevices, useDASDRunningFormatJobs } from "~/queries/dasd";
+import { FormatSummary } from "~/types/dasd";
+
+export default function DASDFormatProgress() {
+  const devices = useDASDDevices();
+  const runningJobs = useDASDRunningFormatJobs();
+
+  const ProgressContent = ({ progress }: { progress: { [key: string]: FormatSummary } }) => (
+    Object.entries(progress).map(([id, { total, step, done }]) => {
+      const device = devices.find((d) => d.id === id);
+
+      return (
+        <Stack hasGutter className="dasd-format-progress">
+          <Progress
+            key={id}
+            size="sm"
+            max={total}
+            value={step}
+            title={`${device.id} - ${device.deviceName}`}
+            measureLocation="none"
+            variant={done ? "success" : undefined}
+          />
+        </Stack>
+      );
+    }));
+
+  return (
+    <Popup title={_("Formatting DASD devices")} isOpen={runningJobs.length > 0} disableFocusTrap>
+      {runningJobs.map((job) => {
+        const progress = job.summary || {};
+
+        if (Object.keys(progress).length === 0) return;
```

Comment:
> Instead of using an `if` block inside of the `map`, I would rather prefer combining a `filter` and a `map`).

NP: I see your point, but probably this was _my fault_. I tend to avoid iterating twice over a same collection. I know, premature optimization.

That said, I'm not against the change you proposed. Most probably, it will bring a more readable code, which is also important.

Thanks for catching it.

Reply:
I was wrong in previous comment since I was thinking about this other `. map` 

https://github.com/openSUSE/agama/pull/1549/files#diff-73b1d201fc2f08a130c9630786cf958273bec4a1e072833919e0f3a541ff1456R87-R94

This one is returning a collection of React.Node or undefined, which is working because React is able to render nothing when `undefined` is given. It's ok since it is a way to ignore them in a _JSX iteration_ but could be done in the way you proposed too.